Shopify/cli
# Fully linked app context

### WHY are these changes introduced?
We need an easy and straightforward way of loading an app with `CurrentAppSchema` and `RemoteAwareSpecifications`
Meaning that is an app that has been linked, the user is authenticated and all extensions were loaded using remote specifications. (i.e. The user is authenticated, the app is valid and up to date)



### WHAT is this pull request doing?
Create a new `app-context` that provides a function to easily load an app using `AppConfigurationState` to validate it.

If the app is not linked, this function will link it.
If the user is not authenticated, this function will force that too.



### How to test your changes?
This new function is used for now in the `build` command as an example. But with the intention of applying it to all commands.

Try running `build` with different scenarios:
- Without being logged in
- With an app that only has `scopes = ""` in the toml (this should force a link)
- With a valid app
- With a config name different from the active one
- With an app using the legacy schema in the toml (it has `client_id = 1231`, important that it must be a number). This will force a re-link.



### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
